packages/api/game/types.go

```diff
@@ -0,0 +1,140 @@
+package game
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+type BaseMessage struct {
+	Event string          `json:"event"`
+	Data  json.RawMessage `json:"data"`
+}
+
+type PlayerJoinData struct {
+	name string
+	x    float32
+	y    float32
+}
+
+type PlayerMoveData struct {
+	id string
+	x  float32
+	y  float32
+}
+
+type IdData struct {
+	id string
+}
+
+type ItemEntityCreateData struct {
+	item string
+	x    float32
+	y    float32
+}
+
+type Game struct {
+	ctx          context.Context
+	connection   *websocket.Conn
+	players      []*Player
+	itemEntities []*ItemEntity
```

Comment:
access to these will need to use a mutex since there will be concurrent access across clients